test/WebJobs.Script.Tests/UtilityTests.cs

```diff
@@ -865,7 +868,78 @@ public void VerifyWorkerIndexingDecisionLogic(bool workerIndexingFeatureFlag, bo
                 testEnv.SetEnvironmentVariable(EnvironmentSettingNames.AzureWebJobsFeatureFlags, ScriptConstants.FeatureFlagEnableWorkerIndexing);
             }
             RpcWorkerConfig workerConfig = new RpcWorkerConfig() { Description = TestHelpers.GetTestWorkerDescription("python", "none", workerIndexingConfigProperty) };
-            bool workerShouldIndex = Utility.CanWorkerIndex(new List<RpcWorkerConfig>() { workerConfig }, testEnv);
+            bool workerShouldIndex = Utility.CanWorkerIndex(new List<RpcWorkerConfig>() { workerConfig }, testEnv, new FunctionsHostingConfigOptions());
+            Assert.Equal(expected, workerShouldIndex);
+        }
+
+        [Theory]
+        [InlineData(true, true, false, "", true)]
+        [InlineData(true, true, true, "NonApp", true)]
+        [InlineData(true, true, true, "AppName", true)]
+        [InlineData(true, true, false, "NonApp", true)]
+        [InlineData(true, true, false, "AppName", true)]
+        public void VerifyWorkerIndexingFeatureFlagTakesPrecedence(bool workerIndexingFeatureFlag, bool workerIndexingConfigProperty, bool enabledHostingConfig, string disabledHostingConfig, bool expected)
+        {
+            VerifyCanWorkerIndexUtility(workerIndexingFeatureFlag, workerIndexingConfigProperty, enabledHostingConfig, disabledHostingConfig, expected);
+        }
+
+        [Theory]
+        [InlineData(true, false, false, "", false)]
+        [InlineData(true, false, true, "NonApp", false)]
+        [InlineData(true, false, true, "AppName", false)]
+        [InlineData(true, false, false, "NonApp", false)]
+        [InlineData(true, false, false, "AppName", false)]
+        [InlineData(true, true, false, "AppName", true)]
+        public void VerifyWorkerConfigTakesPrecedence(bool workerIndexingFeatureFlag, bool workerIndexingConfigProperty, bool enabledHostingConfig, string disabledHostingConfig, bool expected)
+        {
+            VerifyCanWorkerIndexUtility(workerIndexingFeatureFlag, workerIndexingConfigProperty, enabledHostingConfig, disabledHostingConfig, expected);
+        }
+
+        [Theory]
+        [InlineData(false, true, true, "", true)]
+        [InlineData(false, true, true, "NonApp", true)]
+        [InlineData(false, true, false, "NonApp", false)]
+        [InlineData(false, false, false, "NonApp", false)]
```

Comment:
Can we add below few cases as well -
[InlineData(false, false, true, "NonApp", <>)]
[InlineData(false, false, true, "AppName", <>)]
[InlineData(false, false, false, "Appname", <>)]
 [InlineData(false, true, false, "AppName", <>)]
 [InlineData(false, true, true, "AppName", <>)]

Reply:
done